### PR TITLE
Part 2: Generating Insurance Quotes

### PR DESCRIPTION
**Pull request by sajeezy**
### Which part of the onboarding is this for?

- [ ] Part 1: Getting Set Up with Your First Product
- [x] Part 2: Generating Insurance Quotes
- [ ] Part 3: Issuing Policies
- [ ] Part 4: Part 4: Amending Policies with Alteration Hooks
- [ ] Part 5: Lifecycle Hooks and Reactivation
- [ ] Part 6: Scheduled Functions and Anniversary Logic
- [ ] Part 7: Crafting Documents and Documentation
- [ ] Part 8: Building a Claims and Payout Flow

### What has been implemented in this PR?

(The "what" of the PR should be a clearly defined title in present tense i.e "Add Reactivation Logic")

Root Product Builder Fundamentals Course Part 2: Generating Insurance Quotes

### How was testing done?

**Types of tests required**

- [x] Unit test check (tests to be done prior to handing to reviewer)
- [x] API endpoint testing

**Acceptance criteria**
- [x]  A quote can be created via API, or the Root Dashboard, with the correct premium.
- [x]  Quote data validation passes according to the specifications.
- [x]  Unit tests are written for the quote creation.
- [x]  Schemas have been created to get a quote on the Root Dashboard.

A list of tests to perform to pass, split into testing types if required :
- Unit test for Quote hook
  - valid data passes validation
  - returns a suggested premium of <R12.34>
- Using API endpoint test:
  - Was able to create POST request using sandbox
 
**Link to a working quote, application, policy or claim - whichever is relevant to the part**

[Screenshot of Unit tests passing](https://cln.sh/x9yTMtXb)
CleanShot 2023-11-30 at 19 03 48
[Screenshot of Create Quote API response](https://cln.sh/cDVfFts3)
CleanShot 2023-11-30 at 19 16 30@2x

### Authors considerations checklist

- [ ] Has unit tests been written to prove the PR is working?
- [ ] Is this PR being created into the correct branch (main or development)?

### Part Specific Questions

- I struggled with Part 2. It felt like I was on the right track and wasn't able to test my work with ease. Providing a straightforward guide to enable the developer to test their work would have shaved off hours of my time. It could be a simple screencast, loom video detailing that the developer has xyz options available to them.

### CODE REVIEW

File: .root-config.json [hunk 1]
```diff
@@ -75,7 +74,7 @@
       "individualsCustomIdAllowed": false,
       "individualsEmailAllowed": false,
       "companiesAllowed": false,
-      "idCountry": false,
+      "idCountry": "false",
```
Comment (theunisduminy) [inline .root-config.json]: Should not be of type string, but boolean

File: code/unit-tests/01-policy-issue-flow-tests.js
Comment (theunisduminy) [inline code/unit-tests/01-policy-issue-flow-tests.js:1]: These unit tests are missing:

- A payload with valid data passes validateQuoteRequest
- A payload with invalid data fails validateQuoteRequest
- A 20-year-old Tyrannosaurus Rex with R90,000.00 has a premium of R1458.00
- A 36-year-old Velociraptor with R50,000.00 has a premium of R1368.00
- A 16-year-old Brachiosaurus with R65,000.00 has a premium of R1372.80 

The resources folder in the repo has examples of how to do these unit tests. I'll post a video as well to help with this.

File: workflows/quote-schema.json
Comment (theunisduminy) [inline workflows/quote-schema.json:1]: Schemas need to be updated as per the success criteria to be able to get a quote on dashboard. Should reflect your defined validations criteria, so a schema for:

- cover_amount (currency)
- birth_date (date picker)
- species (dropdown)
- start_date (date picker)
- health_checks_updated (checkbox)

**Review by theunisduminy — CHANGES_REQUESTED** [review]
Impressed with how much you figured out just with the docs 😄 

For this part we will need the comprehensive unit tests to see that the premium is correct. The schemas are also not updated yet and that's an important piece for working on dashboard. 

Let me know if you'd like to chat if anything isn't cleat!

File: .root-config.json (hunk 1 above)
Comment (sajeezy) [reply .root-config.json]: every `rp push` adds quotes to this. weird.

File: code/unit-tests/01-policy-issue-flow-tests.js
Comment (sajeezy) [reply code/unit-tests/01-policy-issue-flow-tests.js:1]: I have created these, my mistake
CleanShot 2023-12-01 at 11 26 33

File: workflows/quote-schema.json
Comment (sajeezy) [reply workflows/quote-schema.json:1]: This should be sorted now

**Review by theunisduminy — APPROVED** [review]

**Pull request merged**